### PR TITLE
Fixing extra info on Windows

### LINKED ISSUE #131: Unable to start Sauce Connect after upgrading to version 1.159
**Short summary of what the user sees:**
User is seeing the following error in Sauce Connect logs:
`2022-08-09 15:56:42.220 [17084] [CLI] [fatal] extra info is not valid JSON.`
 
**Exact steps to reproduce the problem:**
Code used for connection
```
if (bSauceLabsExecution) {
    final String testUser = Suite.config.getProperty("sauceLabs.username");
    final String testKey = Suite.config.getProperty("sauceLabs.key");

    tunnelOptions =
            "--tunnel-identifier=" + tunnelId + " --readyfile=connectSet.txt --no-remove-colliding-tunnels";
    /*tunnelOptions =
                "--tunnel-name=" + tunnelId + " --readyfile=connectSet.txt --no-remove-colliding-tunnels";*/

    manager = new SauceConnectFourManager();

    Suite.manager = manager;
    try {
        tunnel = manager.openConnection(testUser,
                testKey, 0, null, tunnelOptions,
                null, false, null);
    } catch (AbstractSauceTunnelManager.SauceConnectException e) {
        e.printStackTrace();
    }
```
 
**Isolation: What must be true for it to happen? When does it not happen?**
User has been using version 1.154.
(Unable to use version 1.158 due to tunnel_name issue.)
 
**Links to Relevant Test Archives or Logs:**
Console output:
`INFO: Launching Sauce Connect 4.8.0 [C:\Users\MO001\sc-4.8.0-win32\bin\sc.exe, -u, ****, -k, ****, -P, 0, --tunnel-identifier=****, --readyfile=connectSet.txt, --no-remove-colliding-tunnels, -v, -l=mySaucelog.txt, --extra-info, {"runner": "jenkins"}]`
 
**What attempts were made to find the root cause so far?**
NA

### PR DESCRIPTION
**Pull request by alexh-sauce**
When extra-info was added, it broke the Windows command line.

Thanks to @aaronmray for the Windows based fix, I had to modify it to work on Linux as well.

Fixes https://github.com/saucelabs/ci-sauce/issues/131

### CODE REVIEW

File: src/test/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManagerTest.java [hunk 1]
```diff
@@ -114,7 +114,12 @@ private Process testDefaultOpenConnection(String logFile, String username) throw
             assertEquals("-P", actualArgs[5]);
             assertEquals(Integer.toString(port), actualArgs[6]);
             assertEquals("--extra-info", actualArgs[7]);
-            assertEquals("{\"runner\": \"jenkins\"}", actualArgs[8]);
+            OperatingSystem operatingSystem = OperatingSystem.getOperatingSystem();
+            if (operatingSystem == OperatingSystem.WINDOWS) {
+                assertEquals("{\\\"runner\\\": \\\"jenkins\\\"}", actualArgs[8]);
+	    } else {
```
Comment (valfirst) [inline src/test/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManagerTest.java:120]:
```suggestion
	        } else {
```

File: src/test/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManagerTest.java [hunk 2]
```diff
@@ -141,7 +146,12 @@ private Process testOpenConnection(String logFile, String username) throws IOExc
...
```
Comment (valfirst) [inline src/test/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManagerTest.java:152]:
```suggestion
	        } else {
```

**Review by valfirst — DISMISSED** [review]

**Comment by frederico-betting** [comment]
Hello, what's the plan to merge this PR? I'm also facing issues on Windows environment without this fix. Thanks.

**Review by amckenzie132 — APPROVED** [review]

File: src/main/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManager.java [hunk 3]
```diff
@@ -255,7 +255,13 @@ protected String[] generateSauceConnectArgs(String[] args, String username, Stri
     }
 
     protected String[] addExtraInfo(String[] args) {
-        String[] result = joinArgs(args, "--extra-info", "{\"runner\": \"jenkins\"}");
+        String[] result;
+        OperatingSystem operatingSystem = OperatingSystem.getOperatingSystem();
+        if (operatingSystem == OperatingSystem.WINDOWS) {
+            result = joinArgs(args, "--extra-info", "{\\\"runner\\\": \\\"jenkins\\\"}");
+        } else {
+            result = joinArgs(args, "--extra-info", "{\"runner\": \"jenkins\"}");
```
Comment (frederico-betting) [inline src/main/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManager.java:263]: Although I'm interested in this fix merged ASAP I'd recommend adding a new test mocking the `getOperatingSystem` response with `Windows` and others. This way you can validate the results and avoid any regression issue. Thoughts @amckenzie132 ?!

Comment (valfirst) [reply src/main/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManager.java:263]: actually CI builds are run on 3 platforms

Comment (frederico-betting) [reply src/main/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManager.java:263]: But this problem happens in Windows environment only. Unix environment (which includes Linux and MacOS) seems to be ok.

Comment (frederico-betting) [reply src/main/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManager.java:263]: Hmmm... those unit tests that were updated are executed in different environment.
In this case, we would have some coverage. Sounds good.

**Review by frederico-betting — APPROVED** [review]

**Comment by alexh-sauce** [comment]
@frederico-betting I've released a new version of the plugin with the fix from ci-sauce:
https://github.com/jenkinsci/sauce-ondemand-plugin/releases/tag/sauce-ondemand-1.209

**Comment by frederico-betting** [comment]
Thanks @alexh-sauce !

**Pull request merged**